app/controllers/votes_controller.rb

```diff
@@ -1,17 +1,18 @@
 class VotesController < ApplicationController
   before_filter :set_question, only: [:show]
+  before_filter :set_question_id, only: [:update]
```

Comment:
This is not wrong and I know it is the rails way but since this is only used in the update function (applies to the other filters below too), I think it is better to keep it inside the respective function. This seems like a premature optimization.

Again, nothing wrong in doing it the way you have it.